rfcs/0030-review-workflow.md

```diff
@@ -0,0 +1,147 @@
+4. It is hard to judge how many PRs actually need to be reviewed at the moment.
+  That kind of statistic would be interesting.
+
+# Detailed design
+[design]: #detailed-design
+
+## PR states
+
+Inspired by the [sage trac](https://trac.sagemath.org/query?desc=1&order=id&group=status&status=!closed&summary=~upgrad&or&status=!closed&summary=~updat),
+I propose that we introduce a set of states a PR might be in. Every PR has to
+be assigned to exactly one of these states and might change between these states
+regularly. The most important states would be `needs:work` and `needs:review`.
+`needs:work` means that the PR is not ready to be merged as-is and needs to be
+improved. `needs:review` means that all the work is done and somebody (usually
+the PR creator) thinks that it should be ready for merge. Reviewers (with or
+without commit access) can then go through the list of `needs:review` PRs and
+review them. They can than either request changes and set the state to
+`needs:work` or merge the PR. In case of review by non-commiters, we could
+introduce an additional state like `needs:merge`. Reviewing and merging those
+PRs should then take little time for the committers and could double as a sort
+of training program for potential new committers.
+
+## Tooling
+
+This workflow is not currently possible due to GitHub's restrictions. Namely
+only people with commit access can set tags on a PR. This can be resolved
+through tooling. Work on this is [in progress](https://github.com/NixOS/ofborg/pull/216).
+
+The tooling should then initialize the PR status on new PRs. In general this
+is going to be `needs:review`. One exception of this are PRs that are work in
+progress (indicated by a `WIP` tag in the title). Those should start at
+`needs:work`. Other exceptions may include RFC type PRs, which may warrant their
+own state (`needs:opinions`?).
+
+To reduce friction for new contributors, the tooling could later be expaned
+to explain this process to first time contributors.
+
+## Advantages
+
+All the above issues should be resolved by this:
+
+1. There is no reason anymore to preferably review recent PRs. Before, recent
+   commit were the only ones where one can be sure that they are not reviewed
+   yet (and thus in a `needs:work` state). Now, reviewers can randomly choose
+   and `needs:review` PR.
+
+2. Similarly, the reviewer may change at any time. If the first reviewer forgets
+   or loses interest in a PR, the PR will just end up in the `needs:review` list
+   again and can easily find a new reviewer.
+
+3. People without commit access can help reviewing commits and lessen the load
+   on committers without putting the reviewed PRs at a disadvantage.
+
+4. We can easily gather statistics. In the long run, we might even implement
+   some sort of automated triage where we regularly ping `needs:work` PRs and
+   close them after the authors went unresponsive for a while. Or we could
+   tag them as `needs:takeover`. But this is out of scope.
```

Comment:
Clearification suggestion: `This is out of scope of this RFC` (also scratch out the "But")